Wikipedia/assets/codemirror/resources/addon/search/search.js

```diff
@@ -30,6 +34,15 @@
         var match = query.exec(stream.string);
         if (match && match.index == stream.pos) {
           stream.pos += match[0].length || 1;
+
+          var fromCursor = cm.getCursor('from')
+          if (stream.lineOracle.line > fromCursor.line || (stream.lineOracle.line == fromCursor.line && stream.start >= fromCursor.ch)) {
+              if (state.initialFocusedMatchIndex == -1) {
```

Comment:
would it make make sense to make it the first check to avoid checking `stream.lineOracle.line > fromCursor.line...` if `initialFocusedMatchIndex` is not -1?

Reply:
Yep agreed, done!